kitsune/sumo/form_fields.py

```diff
@@ -95,29 +95,24 @@ def to_python(self, value):
             else:
                 return []
 
-        # Split names, strip whitespace, and filter out any empty strings
-        # Using set to remove duplicates
-        names = {name.strip() for name in value.split(",") if name.strip()}
+        # Get and strip the input items, turn to list
+        # * "User: user1, Group: group1, User: user2" ->
+        #   ["User: user1", "Group: group1", "User: user2"]
+        parts = (value.strip() for value in value.split(",") if value.strip())
 
-        # Find users and groups in a single query each
-        users_and_groups = list(User.objects.filter(username__in=names)) + list(
-            Group.objects.filter(name__in=names)
-        )
-
-        # Check if all names were found
-        found_names = {
-            obj.username if isinstance(obj, User) else obj.name for obj in users_and_groups
-        }
-        missing_names = names - found_names
+        # Split the parts into key-value pairs
+        # * ["User: user1", "Group: group1", "User: user2"] ->
+        #   [("User", "user1"), ("Group", "group1"), ("User", "user2")]
+        key_value_pairs = (tuple(map(str.strip, part.split(":"))) for part in parts)
```

Comment:
On further review, I realized that this whole chunk could be simplified, I think, to something like:

```python
        key_value_pairs = (tuple(map(str.strip, part.split(":"))) for part in value.split(",") if part.strip())
```

and adding a new comment to fit.

Reply:
This is trading off readability I think, but I used it and just commented it thoroughly.

Reply:
Good point. It might be too much in one line. Feel free to go back to the two-step approach if you think it's more readable.